cmd/init.go

```diff
@@ -981,10 +986,119 @@ var initCmd = &cobra.Command{
+			flagGcpRegion, _ := cmd.Flags().GetString("gcp-region")
+			flagGcpProjectId, _ := cmd.Flags().GetString("gcp-project-id")
+			envGcpRegion := os.Getenv("GCP_REGION")
+			envGcpProjectId := os.Getenv("GCP_PROJECT_ID")
+
+			gcpEnabled := curConfig.GCP.Enabled.ValueOr(true)
+			gcpRegion := curConfig.GCP.Region
+			gcpProjectId := curConfig.GCP.ProjectID
+
+			for {
+				if flagDisableGcp {
+					fmt.Printf("GCP disabled via flags.\n")
+					gcpEnabled = false
+					break
+				}
+
+				gcpEnabled = readBool(
+					"Would you like to enable GCP?",
+					gcpEnabled)
+				if !gcpEnabled {
+					break
+				}
+
+				if flagGcpRegion != "" {
+					fmt.Printf("GCP region specified via flags:\n  %s\n", flagGcpRegion)
+					gcpRegion = flagGcpRegion
+				} else {
+					if gcpRegion == "" && gcpCloudIdent != nil {
+						fmt.Printf("Defaulting to GCP rrgion from self-ident.\n")
+						gcpRegion = gcpCloudIdent.Region
+					}
+					if gcpRegion == "" && envGcpRegion != "" {
+						fmt.Printf("Defaulting to GCP region from environment.\n")
+						gcpRegion = envGcpRegion
+					}
+					if gcpRegion == "" {
+						gcpRegion = cbdcconfig.DEFAULT_GCP_REGION
+					}
+
+					gcpRegion = readString(
+						"What GCP region should we use?",
+						gcpRegion, false)
+				}
+				if gcpRegion == "" {
+					fmt.Printf("The GCP region is required.\n")
+					gcpEnabled = false
+					continue
+				}
+
+				if flagGcpProjectId != "" {
+					fmt.Printf("GCP project id specified via flags:\n  %s\n", flagGcpProjectId)
+					gcpProjectId = flagGcpProjectId
+				} else {
+					if gcpProjectId == "" && envGcpProjectId != "" {
+						fmt.Printf("Defaulting to GCP project id from environment.\n")
+						gcpProjectId = envGcpProjectId
+					}
+					if gcpProjectId == "" && gcpCloudIdent != nil {
+						fmt.Printf("Defaulting to GCP project id from self-ident.\n")
+						gcpProjectId = gcpCloudIdent.ProjectID
+					}
+
+					gcpProjectId = readString(
+						"What GCP project id should we use?",
+						gcpProjectId, false)
+				}
+				if gcpProjectId == "" {
+					fmt.Printf("The GCP project id is required.\n")
+					gcpEnabled = false
+					continue
+				}
+
+				// Validate GCP project ID by attempting to connect to the project
+				creds, err := google.FindDefaultCredentials(ctx, "https://www.googleapis.com/auth/cloud-platform")
+				if err != nil {
+					fmt.Printf("Failed to load GCP credentials: %s\n", err)
+					gcpEnabled = false
+					continue
+				}
+
+				tokenSource := creds.TokenSource
+				client := oauth2.NewClient(ctx, tokenSource)
+
+				computeService, err := compute.NewService(ctx, option.WithHTTPClient(client))
+				if err != nil {
+					fmt.Printf("Failed to create GCP compute client: %s\n", err)
+					gcpEnabled = false
+					continue
+				}
+
+				_, err = computeService.Projects.Get(gcpProjectId).Do()
+				if err != nil {
+					fmt.Printf("Failed to connect to shared GCP project. Please verify that:\n")
```

Comment:
You've done a better job than me making init helpful when things go wrong.  We should mirror this kind of helpful error messaging up to the AWS and Azure steps in the future.